### PR TITLE
Concept Drift Detector

### PR DESCRIPTION
**Pull request by AlifAshrafee**
Drift detector incorporated. But drift detection is not working correctly yet.

### CODE REVIEW

File: utils/drift_detection.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        nn.ReLU(),
+        nn.Flatten(),
+        nn.Linear(2048, encoding_dim)
+    ).to(device).eval()
+
+    # define preprocessing function
+    preprocess_fn = partial(preprocess_drift, model=encoder_net, device=device, batch_size=512)
+
+    # initialise drift detector
+    cd = MMDDrift(ref_data, backend='pytorch', p_val=.05, preprocess_fn=preprocess_fn, n_permutations=100)
```
Comment (jedrzejkozal) [inline utils/drift_detection.py:24]: If I understand the code here correctly, the encoder_net that you pass into preprocess_fn is not trained on incoming data. If we have random weights, the representations returned by the encoder would not be informative for drift detection. I think there are two things we can do about it:
1) Let's try a model that is pretrained on ImageNet instead of this randomly initialized encoder.
2) We can try Uncertainty-based drift detection with the model that was trained so far in the continual training process ( https://docs.seldon.io/projects/alibi-detect/en/stable/cd/background.html#model-uncertainty , https://docs.seldon.io/projects/alibi-detect/en/stable/cd/methods/modeluncdrift.html ). However, the model trained continually is prone to forgetting previous classes, so this could not work.

Comment (AlifAshrafee) [reply utils/drift_detection.py:24]: A new uncertainty-based pre-trained drift detector has been implemented. It works consistently well with all 5 types of drift. However, RandomCrop and Normalize still mess up the detections so I have temporarily disabled them.
